genai-perf/genai_perf/inputs/retrievers/synthetic_data_retriever.py

```diff
@@ -105,44 +112,49 @@ def _generate_stateless_entries(self, use_prefix_prompts: bool) -> List[DataRow]
         data_rows = []
 
         for _ in range(self.config.num_dataset_entries):
-            row = self._create_data_row(use_prefix_prompts)
-            row.images = [
-                self._generate_image() for _ in range(self.config.batch_size_image)
-            ]
+            row = self._create_data_row()
+            row.texts = self._generate_prompts(use_prefix_prompts)
+            row.images = self._generate_images()
             data_rows.append(row)
 
         return data_rows
 
-    def _create_data_row(
-        self, use_prefix_prompts: bool, session_id: str = ""
-    ) -> DataRow:
-        row = DataRow(texts=[], images=[], optional_data={}, payload_metadata={})
+    def _create_data_row(self, session_id: str = "") -> DataRow:
+        row = DataRow()
 
         if session_id:
             row.payload_metadata["session_id"] = session_id
-
-        row.texts = [
-            self._generate_prompt(use_prefix_prompts)
-            for _ in range(self.config.batch_size_text)
-        ]
         return row
 
-    def _generate_prompt(self, use_prefix_prompts: bool) -> str:
-        prompt = SyntheticPromptGenerator.create_synthetic_prompt(
-            self.config.tokenizer,
-            self.config.prompt_tokens_mean,
-            self.config.prompt_tokens_stddev,
-        )
-        if use_prefix_prompts:
-            prefix_prompt = SyntheticPromptGenerator.get_random_prefix_prompt()
-            return f"{prefix_prompt} {prompt}"
-        return prompt
-
-    def _generate_image(self):
-        return SyntheticImageGenerator.create_synthetic_image(
-            image_width_mean=self.config.image_width_mean,
-            image_width_stddev=self.config.image_width_stddev,
-            image_height_mean=self.config.image_height_mean,
-            image_height_stddev=self.config.image_height_stddev,
-            image_format=self.config.image_format,
-        )
+    def _generate_prompts(self, use_prefix_prompts: bool) -> List[str]:
+        prompts = []
+        for _ in range(self.config.batch_size_text):
+            prompt = SyntheticPromptGenerator.create_synthetic_prompt(
+                self.config.tokenizer,
+                self.config.prompt_tokens_mean,
+                self.config.prompt_tokens_stddev,
+            )
+            if use_prefix_prompts:
+                prefix_prompt = SyntheticPromptGenerator.get_random_prefix_prompt()
+                prompt = f"{prefix_prompt} {prompt}"
+
+            prompts.append(prompt)
+        return prompts
+
+    def _generate_images(self) -> List[str]:
+        """
+        Generate synthetic images if the image width and height are specified.
+        """
+        images = []
+        if self._include_image:
```

Comment:
i am not sure so i want your opinion here.
why do the if check here as opposed to outside the method call in _generate_prompts()

Reply:
I just think it's easier to avoid mistakes when the callee takes care of these simple checks. Let's say in the future, we want to call `self._generate_images` in `_generate_multi_turn_sessions`, or somewhere else in the code. Every time we call this function, the caller doesn't need to remember to add `if` statement to avoid unnecessary image data generation. Same for audio data generation once we have one.

Reply:
I think you are highlighting the tradeoffs well. 
I went back and forth on whether the if should be in the caller or callee. There are reasons for either approach.
Ultimately, i think its a style conclusion, and we should just try to remain consistent.